crocket/src/main/java/com/crocket/CroquetView.java

```diff
@@ -1,32 +1,71 @@
 package com.crocket;
 import javax.swing.*;
 
+import java.awt.Component;
 import java.awt.Dimension;
 import java.awt.Toolkit;
 
+/*
+ * Class: CroquetView
+ * 
+ * This class extends JFrame and represents the main window of the game. It is responsible for 
+ * rendering the game state to the user.
+ * 
+ * CroquetView is implemented as a Singleton, ensuring that only one instance of CroquetView exists
+ * in the application at any time. This is done to maintain a consistent view state throughout
+ * the application.
+ * 
+ * Methods:
+ * - drawLevel(LevelView drawLevel): Adds the provided LevelView to the JFrame, effectively drawing the level.
+ * - removeAllLevelComponents(): Removes all LevelView components from the JFrame.
+ * - getInstance(): Returns the singleton instance of CroquetView.
+ * 
+ */
+
 public class CroquetView extends JFrame{
-    JFrame frame;
-    Dimension screen = Toolkit.getDefaultToolkit().getScreenSize();
-    DrawEntity entity = new DrawEntity();
-    DrawField field = new DrawField();
-    ImageIcon img = new ImageIcon("crocket/assets/textures/JFrame_Icon.jpg");
-    String title = "Krocket";
-    boolean visibility = true;
+    private static CroquetView INSTANCE;
+    private ImageIcon img = new ImageIcon("crocket/assets/textures/JFrame_Icon.jpg");
+    private String title = "Krocket";
+    public DrawEntity entity = new DrawEntity();
+
+    public void drawLevel(LevelView level){
+        removeAllLevelComponents();
+        add(level);
+        level.add(entity);
+    }
+
+    private void removeAllLevelComponents(){
+        Component[] components = this.getComponents();
+        for (Component component : components) {
+            if (component instanceof LevelView) {
```

Comment:
Is there maybe a way to do this without using `instanceof`?

Reply:
Used getClass instead but might still be ugly. Killing polymorphism

Reply:
Instanceof tends to be a sign of bad or wrong use of polymorphism

Reply:
The check you changed to basically does the same thing. Is there a way to to set this up where we don't have to check the class of something?

Reply:
No it just clears the frame when we add a level